crates/miden-testing/src/tx_context/context.rs

```diff
@@ -55,9 +55,9 @@ pub struct TransactionContext {
     pub(super) expected_output_notes: Vec<Note>,
     pub(super) foreign_account_inputs: BTreeMap<AccountId, AccountInputs>,
     pub(super) tx_args: TransactionArgs,
-    pub(super) tx_inputs: TransactionInputs,
+    pub(super) tx_inputs: TransactionKernelInputs,
     pub(super) mast_store: TransactionMastStore,
-    pub(super) advice_inputs: AdviceInputs,
+    pub(super) advice_inputs: AdviceInputs, // TODO: unused, something must be wrong?
```

Comment:
@bobbinth @PhilippGackstatter I am confused about the `Option<AdviceInputs>` vs these `AdviceInputs`. Right now I have the former being passed into `TransactionKernelInputs::new()` but I have probably done something wrong to leave this one unused.

Reply:
I have left TODO statements to cover the places where there is confusion around the advice inputs

Reply:
> I this also applicable outside of testing scenarios?

Probably less so, yes. I extrapolated a bit too much from the tx context which is the only non-trivial user of the `TransactionExecutor` API we have in base.